jsoncache.js

```diff
@@ -4,11 +4,13 @@ const http = require('http');
 const https = require('https');
 
 class JSONCache {
-  constructor(url, timeout, promiseLib = Promise) {
+  constructor(url, timeout, promiseLib = Promise, maxRetry) {
```

Comment:
should be `constructor(url, timeout, promiseLib = Promise, maxRetry = 30) {` so that if none is specified, the assignment in the constructor doesn't need extra logic.